File: platform-java/services/trust-service/src/main/java/com/grassland/trust/judge/Judge.java
package com.grassland.trust.judge;

import java.time.Instant;

/**
 * 审判官（草场 Epic 6 Slice 6C / HLD 3.1「符合条件的推荐官」+ §5.5 + 任务书 #74 卡 E）。
 *
 * <p>{@code accountId} 与 identity 账号同空间（database-per-service 无 FK）；{@code organizationId} 可空=平台级审判官；
 * {@code eligibilityTier} 来自 marketplace 有效等级；只有 active、{@code opsAdmitted} 且满足 V14 资格谓词
 * （Lv5 直入 / Lv4+考试及格见习）且未挂起才可被抽签。同组织或显式 {@code judge_conflict} 记录构成利益冲突，
 * 抽面板时排除。
 *
 * <p>卡 E 新列：{@code examPassedAt}=准入考试及格时刻；{@code admissionLevel}=full/probation（见习每面板
 * ≤2 席、累计 10 轮投票无异常自动转正）；{@code suspendedUntil}/{@code suspensionReason}=运营确认挂起
 * （90 天窗口弃权率 >40% 的一键建议+人工确认，v1 无自动挂起定时器）。
 */
public record Judge(
        String id,
        String accountId,
        String organizationId,
        int eligibilityTier,
        boolean active,
        boolean opsAdmitted,
        long version,
        Instant opsAdmittedAt,
        String opsAdmittedBy,
        Instant createdAt,
        Instant examPassedAt,
        String admissionLevel,
        Instant probationSince,
        Instant suspendedUntil,
        String suspensionReason) {

    /** 既有 10 参调用方兼容（#74 之前）：无考试/见习/挂起信息。 */
    public Judge(String id, String accountId, String organizationId, int eligibilityTier, boolean active,
                 boolean opsAdmitted, long version, Instant opsAdmittedAt, String opsAdmittedBy,
                 Instant createdAt) {
        this(id, accountId, organizationId, eligibilityTier, active, opsAdmitted, version, opsAdmittedAt,
                opsAdmittedBy, createdAt, null, "full", null, null, null);
    }

    /** 是否见习审判官（admission_level=probation）。 */
    public boolean isProbation() {
        return "probation".equals(admissionLevel);
    }

    /** 当前是否处于挂起期（suspended_until 未到）。 */
    public boolean suspendedNow() {
        return suspendedUntil != null && Instant.now().isBefore(suspendedUntil);
    }
}

File: src/types/grassland/judge.ts
// ---------- trust：审判官池 + 投票（+ 任务书 #74 卡 E 考试/见习/考核）----------

import type { AdjudicationSnapshot } from './dispute'

/** 审判官入池记录。active=false 为已退池（软删，保留历史面板/投票完整性）。 */
export interface Judge {
  id: string
  accountId: string
  /** 归属组织；null = 平台级审判官。抽签时排除与争议同组织者。 */
  organizationId: string | null
  /** 报名时的有效等级快照；抽签和运营授权时仍会实时复验。 */
  eligibilityTier: number
  active: boolean
  opsAdmitted: boolean
  version: number
  opsAdmittedAt: string | null
  opsAdmittedBy: string | null
  /** 任务书 #74 卡 E：准入考试 / 见习标识 / 挂起状态（审判台展示）。 */
  admissionLevel?: 'full' | 'probation'
  probation?: boolean
  examPassedAt?: string | null
  suspendedNow?: boolean
  suspendedUntil?: string | null
  createdAt: string | null
}

/** 卡 E：准入审计动作（考试及格=probation、转正=promoted、挂起/恢复）。 */
export interface JudgeAdmissionAudit {
  id: number
  action: 'granted' | 'revoked' | 'probation' | 'promoted' | 'suspended' | 'reinstated'
  actorAccountId: string
  reason: string
  previousVersion: number
  newVersion: number
  createdAt: string | null
}

export interface AdminJudge extends Judge {
  audit?: JudgeAdmissionAudit[]
  probationSince?: string | null
  suspensionReason?: string | null
}

export interface AdminJudgePage {
  items: AdminJudge[]
  nextCursor: string | null
  hasMore: boolean
}

export interface UpdateJudgeAdmissionInput {
  admitted: boolean
  expectedVersion: number
  reason: string
}

/** 投票选择。abstain 不计入任一方多数。 */
export type VoteChoice = 'for_merchant' | 'for_recommender' | 'abstain'

export interface JudgeVote {
  disputeId: string
  round: number
  vote: VoteChoice
  rationale: string | null
  votedAt: string | null
  tallies: AdjudicationSnapshot['tallies']
}

// ---------- 任务书 #74 卡 G：脱敏判例库 ----------

/** 判例（构造性脱敏：无 org/account/金额字段）。 */
export interface PrecedentCase {
  id: string
  disputeId: string
  taskType: string | null
  taskPlatform: string | null
  disputeKind: string | null
  /** 争议焦点（kind+platform+reason 前 80 字）。 */
  focus: string | null
  /** 双方主张摘要（脱敏证据元信息拼接）。 */
  claimsSummary: string | null
  decision: string | null
  /** 终局经由：panel=面板多数 / cs=客服终裁 / retrial=发回重审后再终局。 */
  finalVia: 'panel' | 'cs' | 'retrial' | string
  /** 各轮投票分布 JSON 字符串：[{forMerchant, forRecommender, abstain, matchedPlatformCount}]。 */
  voteSummary: string | null
  /** 终局轮每票理由摘要数组 JSON 字符串（不含审判官账号）。 */
  rationaleDigest: string | null
  createdAt: string | null
}

// ---------- 任务书 #74 卡 E：准入考试 ----------

/** 考试题（用户端出题响应不含 answerIndex）。 */
export interface JudgeExamQuestion {
  id: string
  category: string
  question: string
  options: string[]
  answerIndex?: number
  active?: boolean
  version?: number
  createdAt?: string | null
}

export interface JudgeExamAttempt {
  id: string
  accountId: string
  score: number
  passed: boolean
  answers: string | null
  createdAt: string | null
}

/** 考核看板行（90 天窗口实时聚合）。 */
export interface JudgeAssessmentRow {
  accountId: string
  assigned: number
  voted: number
  abstained: number
  abstainRate: number
  /** 弃权率 >40% 且分配 ≥5 次 → 建议暂停（运营确认制）。 */
  suggestSuspension: boolean
}
